File: gm-utils/src/main/java/src/commom/utils/object/Safe.java
package src.commom.utils.object;

import gm.utils.lambda.F0;
import gm.utils.lambda.F1;

public class Safe {

	public static <OUT,IN> OUT get(IN o, F1<IN,OUT> getter, F0<OUT> other) {
		if (Null.is(o)) {
			return other.call();
		}
		return getter.call(o);
	}

	public static <OUT,IN> OUT safeNullPointer(IN o, F1<IN,OUT> getter, F0<OUT> other) {
		if (Null.is(o)) {
			return other.call();
		}
		try {
			return getter.call(o);
		} catch (Exception e) {
			return other.call();
		}
	}

}

File: gm-utils/js/src/commom/utils/object/Safe.js
import Null from './Null';

export default class Safe {

	static get(o,getter, other) {
		if (Null.is(o)) {
			return other();
		}
		return getter(o);
	}

	static safeNullPointer(o,getter, other) {
		if (Null.is(o)) {
			return other();
		}
		try {
			return getter(o);
		} catch (e) {
			return other();
		}
	}

}
